source/D2Common/src/D2StatList.cpp

```diff
@@ -2702,58 +1776,35 @@ void __stdcall D2Common_10533(D2UnitStrc* pUnit)
 //D2Common.0x6FDB8900
 void __stdcall D2Common_STATES_ToggleState_6FDB8900(D2UnitStrc* pUnit, int nState, BOOL bSet)
 {
-	D2StatesTxt* pStatesTxtRecord = NULL;
-	BOOL bIsSetAlready = FALSE;
-
-	if (pUnit && pUnit->pStatListEx && (pUnit->pStatListEx->dwFlags & 0x80000000) != 0)
+	
+	if (pUnit && pUnit->pStatListEx && STATLIST_IsExtended(pUnit->pStatListEx))
 	{
-		bIsSetAlready = pUnit->pStatListEx->StatFlags[nState >> 5] & gdwBitMasks[nState & 31];
+		const uint32_t dwStateBitMask = gdwBitMasks[nState % 32];
```

Comment:
In my opinion this is way more confusing than it was before due to all the bit operations and the additional usage of a reference

Reply:
I was not really satisfied with it either, I think a helper function that sets a given bit and returns if it changed (to put in [D2BitManip.h](source/Fog/include/D2BitManip.h) ?) would make this a lot cleaner ?
It seems the pattern comes pretty often

Reply:
I haven't see it often yet, but I would be fine with that